File: Preview/yarn/listbuilds/listbuilds.js
'use strict';

async function listBuilds() {
  const gcpMetadata = require('gcp-metadata');
  const projectId = await gcpMetadata.project('project-id');

  const {CloudBuildClient} = require('@google-cloud/cloudbuild');
  const cb = new CloudBuildClient();

  const [result] = await cb.listBuilds({projectId});

  var i = 0;
  for (i = 0; i < 10 && i < result.length; i++) {
    console.log("Build %s %s", result[i].id, result[i].status)
  }
  console.log("Listed %d builds for project %s.", i, projectId)
}

listBuilds().catch(console.error);

File: Preview/mvn/listbuilds/src/main/java/ListBuilds.java
import java.io.IOException;

import com.google.cloud.ServiceOptions;
import com.google.cloud.devtools.cloudbuild.v1.CloudBuildClient;
import com.google.cloudbuild.v1.Build;

public class ListBuilds {
    public static void main(String[] args) {
		int limit = 10;
		if (args.length == 1) {
			limit = Integer.parseInt(args[0]);
		}
		String projectId = ServiceOptions.getDefaultProjectId();

		int n = 0;
        try (CloudBuildClient cb = CloudBuildClient.create()) {
            for (Build b : cb.listBuilds(projectId, "").iterateAll()) {
                System.out.printf("Build: %s %s\n", b.getId(), b.getStatus());
				n++;
				if (n==limit) {
					break;
				}
            }
        } catch (IOException e) {
			System.err.printf("ERROR: %s.\n", e);
		}
        System.out.printf("Listed %d builds for project %s.\n", n, projectId);
    }
}
